The GMP include is hardcoded to /usr/include causing obvious problems.

From classpath CVS:
http://cvs.savannah.gnu.org/viewvc/classpath/configure.ac?root=classpath&r1=1.245&r2=1.246

[Gustavo: ditch -L or libtool complains loudly]

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

--- a/configure.ac	2010/01/30 01:33:50	1.245
+++ b/configure.ac	2010/05/04 12:29:23	1.246
@@ -752,10 +752,32 @@
   dnl __gmpz_mul_si for earlier versions (>= 3.1).
   dnl IMPORTANT: if you decide to look for __gmpz_combit, don't forget to
   dnl change the name of the corresponding ac_ variable on lines 860...
+  AC_ARG_WITH(gmp, [  --with-gmp=PATH         specify prefix directory for the installed GMP package.
+                          Equivalent to --with-gmp-include=PATH/include
+                          plus --with-gmp-lib=PATH/lib])
+  AC_ARG_WITH(gmp_include, [  --with-gmp-include=PATH specify directory for installed GMP include files])
+  AC_ARG_WITH(gmp_lib, [  --with-gmp-lib=PATH     specify directory for the installed GMP library])
+
   if test "x${COMPILE_GMP}" = xyes; then
+    AC_MSG_CHECKING([for GMP directories])
+    if test "x$with_gmp" != x; then
+       gmplibs="-lgmp"
+       gmpinc="-I$with_gmp/include"
+       if test "x$with_gmp_include" != x; then
+       	  gmpinc="-I$with_gmp_include $gmpinc"
+       fi
+       if test "x$with_gmp_lib" != x; then
+       	  gmplibs="$gmplibs"
+       fi
+    else
+        with_gmp="/usr"
+	gmplibs="-lgmp"
+    	gmpinc="-I/usr/include"
+    fi
+    AC_MSG_RESULT([prefix=$with_gmp, libs=$gmplibs, inc=$gmpinc])
     AC_CHECK_LIB(gmp, __gmpz_mul_si,
-      [GMP_CFLAGS=-I/usr/include
-       GMP_LIBS=-lgmp ],
+      [GMP_CFLAGS="$gmpinc"
+       GMP_LIBS="$gmplibs" ],
       [GMP_CFLAGS=
        GMP_LIBS= ])
     AC_SUBST(GMP_CFLAGS)
